spec/requests/notifications_spec.rb

```diff
@@ -52,15 +54,18 @@
           end
 
           context 'and the requested format is html' do
-            let(:format) { 'html' }
+            let(:headers) { { 'HTTP_REFERER' => previous_page } }
 
             it 'redirects the user back to where they were before' do
               expect(response).to redirect_to previous_page
             end
           end
 
           context 'and the requested format is json' do
-            let(:format) { 'json' }
+            let(:headers) do
```

Comment:
I feel like this headers let block is repeated way too many times. Could be extracted to a higher context and shared. 
Make the accept block `'ACCEPT' => format` and define `let(:format)` where needed

Reply:
In few context blocks, we need headers to have JSON format. And for a few HTML. That's the reason why headers are duplicated.

Reply:
My recommendation is to reduce the amount of duplicated long blocks of code by moving the headers block to a higher context and defining the format where needed
```
context 'when user is signed in' do
  let(:headers) do
    { 
      'HTTP_REFERER' => previous_page,
      'ACCEPT' => format
    }
  end

  context '...' do
    let(:format) { 'application/json' }
  end

  context '...' do
    let(:format) { 'text/html' }
  end
end
```

Reply:
okay thank you. Will do this change.

Reply:
I have pushed above changes. Could you please review.